src/frontend/src/icp-eth/services/custom-token.services.ts

```diff
@@ -0,0 +1,107 @@
+import type { OptionErc20Token } from '$eth/types/erc20';
+import { loadCustomTokens } from '$icp/services/icrc.services';
+import type { IcrcCustomToken } from '$icp/types/icrc-custom-token';
+import { setCustomToken as setCustomTokenasApi } from '$lib/api/backend.api';
+import { busy } from '$lib/stores/busy.store';
+import { i18n } from '$lib/stores/i18n.store';
+import { toastsError } from '$lib/stores/toasts.store';
+import type { OptionIdentity } from '$lib/types/identity';
+import type { Token } from '$lib/types/token';
+import type { Identity } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+import { assertNonNullish, isNullish, toNullable } from '@dfinity/utils';
+import { get } from 'svelte/store';
+
+/**
+ * When a user converts a ckERC20 token to an ERC20 twin token, the UI needs information about the counterpart token (ckERC20).
+ * For example, the user might have an ERC20 token (e.g., USDC) enabled, but its token counterpart (e.g., ckUSDC) disabled.
+ * As a result, the UI might lack necessary information.
+ * Therefore, this function aims to enable the CK token if the user has only enabled the counterpart token.
+ *
+ * @param {Object} params - The parameters for the function.
+ * @param {Erc20UserToken[]} params.icrcCustomTokens - The list of user's ICRC tokens.
+ * @param {Token} params.sendToken - The token to be sent.
+ * @param {OptionIdentity} params.identity - The user's identity.
+ * @returns {Promise<{ result: 'loaded' | 'skipped' | 'error' }>} The result of the operation.
+ */
+export const autoLoadCustomToken = async ({
+	icrcCustomTokens,
+	sendToken,
+	identity
+}: {
+	icrcCustomTokens: IcrcCustomToken[];
+	sendToken: Token;
+	identity: OptionIdentity;
+}): Promise<{ result: 'loaded' | 'skipped' | 'error' }> => {
+	if (sendToken.standard !== 'erc20') {
+		return { result: 'skipped' };
+	}
+
```

Comment:
In the other loader, here a test is finding place:

```
const twinToken = (sendToken as OptionIcCkToken)?.twinToken;

	if (twinToken?.standard !== 'erc20') {
		return { result: 'skipped' };
	}

```

Shouldn't we mirror it for icrc?

Reply:
Not really, sure, I need your opinion:

We are not getting directly the twin token (like for ERC20), but the twin token symbol. Then we search for the ck-token among ICRC tokens. So, theoretically, they will always have the correct standard, no? Or shall I check either way for good measure?

Reply:
Better to put it... included